packages/cli/src/commands/doctor/healthchecks/reactNativeGlobal.ts

```diff
@@ -0,0 +1,155 @@
+import {HealthCheckInterface} from '../types';
+import fs from 'fs';
+import execa from 'execa';
+
+import globalDirectories from 'global-dirs';
+import isPathInside from 'is-path-inside';
+import {logger} from '@react-native-community/cli-tools';
+import {logError} from './common';
+import path from 'path';
+
+const label = 'react-native/react-native-cli globally installed';
+
+const resolveGlobalYarnPackage = moduleId => {
+  try {
+    const modulePath = require.resolve(
+      path.join(globalDirectories.yarn.packages, moduleId),
+    );
+    return modulePath;
+  } catch (error) {
+    return '';
+  }
+};
+
+const resolveGlobalNpmPackage = moduleId => {
+  try {
+    const modulePath = require.resolve(
+      path.join(globalDirectories.npm.packages, moduleId),
+    );
+    return modulePath;
+  } catch (error) {
+    return '';
+  }
+};
+
+const checkGlobalPaths = moduleId => {
+  return !!(
+    (resolveGlobalYarnPackage(moduleId) || resolveGlobalNpmPackage(moduleId)) &&
+    true
+  );
+};
+
+const checkGlobalInstall = () => {
+  let reactNativeCLIGlobal;
+  let reactNativeGlobal;
+
+  try {
+    reactNativeGlobal = checkGlobalPaths('react-native');
+  } catch (error) {
+    return null;
+  }
+
+  try {
+    reactNativeCLIGlobal = checkGlobalPaths('react-native-cli');
+    logger.log('******Found r-n-cli: ' + reactNativeCLIGlobal);
+  } catch (error) {
+    return null;
+  }
```

Comment:
This can be reduced to something like:

```typescript
// Can be at the top of the file
const librariesToCheck = ['react-native', 'react-native-cli'];

const checkGlobalInstallations = () => {
  const librariesGloballyInstalled = librariesToCheck.map(library =>
    checkGlobalPaths(library),
  );

  // This way `librariesGloballyInstalled` will hold an array of paths
  // then you can loop through it to remove the libraries
  return librariesGloballyInstalled;
}
```